### PR TITLE
MAT-5357: New Measure's Start and End set to first and last second of the day

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-5357](https://jira.cms.gov/browse/MAT-5357)
(Optional) Related Tickets:
https://github.com/MeasureAuthoringTool/madie-measure/pull/313

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

**Review by sb-prateekkeerthi — DISMISSED** [review]
If you are doing this change, you might also want to apply the same changes in measure edit details page because even there you can set the measurement periods

**Comment by jacobmainzer-sb** [comment]
The time being sent is the offset of the local time at midnight in UTC.  So the datepicker is setting the JS date to midnight local time, and then converts it to an ISO string in UTC.  The problem is when this code switches the string to midnight UTC, the datepicker on the  measure edit details page converts the date into local time again, which for the US would result in the date being displayed as one day earlier.

The header on the measure also shows the incorrect date (one day earlier).

File: src/components/NewMeasure/CreateNewMeasureDialog.tsx [hunk 1]
```diff
@@ -380,7 +391,18 @@ const CreateNewMeasureDialog = ({ open, onClose }) => {
...
-              formik.setFieldValue("measurementPeriodEnd", endDate);
+              const utcEndDate = new Date(
+                Date.UTC(
+                  endDate.getFullYear(),
+                  endDate.getMonth(),
+                  endDate.getDate(),
+                  0,
+                  0,
+                  0,
+                  0
```
Comment (jacobmainzer-sb) [inline src/components/NewMeasure/CreateNewMeasureDialog.tsx]: Based on the Slack discussion, shouldn't this be set to the end of the entered day?

Comment (ethankaplan) [reply src/components/NewMeasure/CreateNewMeasureDialog.tsx:402]: I forgot to save before committing...

Comment (ethankaplan) [reply src/components/NewMeasure/CreateNewMeasureDialog.tsx]: Oops, posted for review before I pushed

**Comment by jacobmainzer-sb** [comment]
What happens when an Eastern Time user creates a measure and a Pacific Time user views the same measure?  Or vice versa?

**Comment by ethankaplan** [comment]
> What happens when an Eastern Time user creates a measure and a Pacific Time user views the same measure? Or vice versa?

We talked about it on slack, the goal of this ticket is just the times, not the time zones, unless I read the conversation wrong
https://semanticbits.slack.com/archives/C02GDA28AVC/p1676656898302929

**Comment by jacobmainzer-sb** [comment]
Yes, that seemed to be the result.  But by moving the times to the start and end of the day, it's going to cause a more visible timezone problem to end users.  I want to make sure we are OK with having users seeing different start and end dates and any side effects that could cause.
